### ISSUE TITLE
On windows platform teardown steps fails to delete context because files are still in use

### ISSUE THREAD

**Issue by prietyc123** [openshift/odo#3371]
/kind bug

## What versions of software are you using?

**Operating System:**
Win10

**Output of `odo version`:**
master

## How did you run odo exactly?

Running `make test-cmd-devfile-watch` on windows powershell terminal.

## Actual behavior

```
> make test-cmd-devfile-watch
ginkgo  -randomizeAllSpecs -slowSpecThreshold=120 -timeout 7200s -nodes=1 -focus="odo devfile watch command tests" tests/integration/devfile/
Running Suite: Devfile Suite
============================
Random Seed: 1592388378 - Will randomize all specs
Will run 6 of 60 specs
SSSSSSSSSS+SSS+SSSSSSSSSSSSSSSSSSSSSSSS+SSSSS+SSSSSSSSRunning command:  [odo watch  --context C:\Users\Admin\AppData\Local\Temp\647821543]
Created dir: C:\Users\Admin\AppData\Local\Temp\647821543
Setting KUBECONFIG=C:\Users\Admin\AppData\Local\Temp\647821543\config
Creating a new project: dzqnphrivh
Running odo.exe with args [odo project create dzqnphrivh -w -v4]
[odo]  -  Waiting for project to come up  ...
[odo] I0617 10:06:37.483866    7540 occlient.go:543] Status of creation of project dzqnphrivh is Active
[...]
[odo]  V  Project 'dzqnphrivh' is ready for use
[odo]  V  New project created and now using project: dzqnphrivh
[odo] I0617 10:06:37.562831    7540 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
Current working dir: C:\Users\Admin\go\src\github.com\openshift\odo\tests\integration\devfile
Setting current dir to: C:\Users\Admin\AppData\Local\Temp\647821543
Running odo.exe with args [odo preference set Experimental true]
[odo] I0617 10:06:37.670415    2920 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
[odo] Global preference was successfully updated
[...]
Running odo.exe with args [odo create nodejs --project dzqnphrivh lkvpwi]
[odo] I0617 10:06:37.838760   11028 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
[odo] Experimental mode is enabled, use at your own risk
[odo]
[odo] I0617 10:06:37.880608   11028 util.go:399] path ./.odo/config.yaml doesn't exist, skipping it
[odo] I0617 10:06:37.880608   11028 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
[odo] I0617 10:06:37.883330   11028 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
[odo] I0617 10:06:39.471178   11028 util.go:399] path devfile.yaml doesn't exist, skipping it
[odo] I0617 10:06:39.471178   11028 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
[odo] I0617 10:06:39.471178   11028 util.go:399] path devfile.yaml doesn't exist, skipping it
[odo] Validation
[odo]  -  Checking devfile compatibility  ...
[odo]  -  Creating a devfile component from registry: DefaultDevfileRegistry  ...
I0617 10:06:39.471178   11028 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
 V  Checking devfile compatibility [0ns]
 V  Creating a devfile component from registry: DefaultDevfileRegistry [0ns]
[odo]  -  Validating devfile component  ...
[odo] I0617 10:06:39.472269   11028 util.go:399] path .odo\env\env.yaml doesn't exist, skipping it
[odo] I0617 10:06:39.472269   11028 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
 V  Validating devfile component [1ms]
[odo] I0617 10:06:39.472269   11028 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
[odo] I0617 10:06:39.472269   11028 util.go:399] path devfile.yaml doesn't exist, skipping it
[odo]
[odo] Please use `odo push` command to create the component with source deployed
[odo] I0617 10:06:39.542637   11028 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
Running odo.exe with args [odo push --devfile devfile.yaml --project dzqnphrivh]
[odo] I0617 10:06:39.677232    5916 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
[odo] I0617 10:06:39.720051    5916 preference.go:188] The path for preference file is C:\Users\Admin\AppData\Local\Temp\647821543\config.yaml
[...]
[odo]  V  Changes successfully pushed to component
[odo] I0617 10:06:50.421320    5916 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
Deleting dir: C:\Users\Admin\AppData\Local\Temp\647821543\abcd
Replacing "Hello" with "Hello odo" in C:\Users\Admin\AppData\Local\Temp\647821543\app\app.js
Running oc.exe with args [oc get pods --namespace dzqnphrivh --show-labels]
[oc] NAME                     READY   STATUS    RESTARTS   AGE   LABELS
[oc] lkvpwi-576f7f6dc-p4h9m   1/1     Running   0          31s   component=lkvpwi,pod-template-hash=576f7f6dc
Running oc.exe with args [oc exec lkvpwi-576f7f6dc-p4h9m --namespace dzqnphrivh -- ls -lai /projects/nodejs-web-app]
[oc] total 44
[oc] 281176798 drwxrwsrwx.  6 1000910000 1000910000   190 Jun 17 10:07 .
[oc] 188767190 drwxrwsrwx.  3 root       1000910000    28 Jun 17 10:06 ..
[oc] 140611490 drwxrwsrwx.  2 1000910000 1000910000     6 Jun 17 10:07 .abc
[oc] 281176799 -rw-rw-rw-.  1 1000910000 1000910000    25 Jun 17 10:06 .gitignore
[oc] 283235153 drwxrwsrwx.  3 1000910000 1000910000    44 Jun 17 10:06 .odo
[oc] 281176809 -rw-rw-rw-.  1 1000910000 1000910000     0 Jun 17 10:07 a.txt
[oc] 287485128 drwxrwsrwx.  2 1000910000 1000910000    20 Jun 17 10:07 app
[oc] 281176800 -rw-rw-rw-.  1 1000910000 1000910000  5369 Jun 17 10:06 config
[oc] 281176801 -rw-rw-rw-.  1 1000910000 1000910000    80 Jun 17 10:06 config.yaml
[oc] 281176802 -rw-rw-rw-.  1 1000910000 1000910000  1162 Jun 17 10:06 devfile.yaml
[oc] 310393695 drwxrwsrwx. 52 1000910000 1000910000  4096 Jun 17 10:06 node_modules
[oc] 281176808 -rw-rw-rw-.  1 1000910000 1000910000 14663 Jun 17 10:06 package-lock.json
[oc] 281176803 -rw-rw-rw-.  1 1000910000 1000910000   279 Jun 17 10:06 package.json
Deleting project: dzqnphrivh
Running odo.exe with args [odo project delete dzqnphrivh -f]
[odo] I0617 10:07:12.157937    9372 application.go:49] Unable to list Service Catalog instances: unable to list ServiceInstances: the server could not find 
the requested resource (get serviceinstances.servicecatalog.k8s.io)
[odo]  !  Warning! Projects are deleted from the cluster asynchronously. Odo does its best to delete the project. Due to multi-tenant clusters, the project 
may still exist on a different node.
[odo] I0617 10:07:12.231407    9372 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
[odo]  V  Deleted project : dzqnphrivh
Setting current dir to: C:\Users\Admin\go\src\github.com\openshift\odo\tests\integration\devfile
Deleting dir: C:\Users\Admin\AppData\Local\Temp\647821543
------------------------------
+ Failure in Spec Teardown (AfterEach) [36.450 seconds]
odo devfile watch command tests
C:/Users/Admin/go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:15
  when executing odo watch after odo push [AfterEach]
  C:/Users/Admin/go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:83
    should listen for file changes
    C:/Users/Admin/go/src/github.com/openshift/odo/tests/integration/devfile/cmd_devfile_watch_test.go:84
    Unexpected error:
        <*os.PathError | 0xc000405860>: {
            Op: "remove",
            Path: "C:\\Users\\Admin\\AppData\\Local\\Temp\\647821543\\a.txt",
            Err: 0x20,
        }
        remove C:\Users\Admin\AppData\Local\Temp\647821543\a.txt: The process cannot access the file because it is being used by another process.
    occurred
    C:/Users/Admin/go/src/github.com/openshift/odo/tests/helper/helper_filesystem.go:28
```

## Expected behavior

Running [devfile watch file](https://github.com/openshift/odo/blob/master/tests/integration/devfile/cmd_devfile_watch_test.go#L82) on windows but [teardown step](https://github.com/openshift/odo/blob/master/tests/integration/devfile/cmd_devfile_watch_test.go#L46) throws error due to file is being used by another process .

**Comment by prietyc123** [comment]
Same failure occurrence seen with `make test-cmd-docker-devfile-watch`

**Comment by jgwest** [comment]
Hey all, I will have a fix for this as part of https://github.com/openshift/odo/issues/3435

**Comment by prietyc123** [comment]
> Hey all, I will have a fix for this as part of #3435

Ahh.. somehow we missed this comment and @anandrkskd was working on this. Anyway thanks for the change 👍
